File: axion/project 3/flutter_application_1/lib/models/allergy_intolerance.dart
class AllergyIntolerance {
  final String patient;
  final String code;
  final String display;
  final String timestamp;
  final String? criticality;
  final String? severity;
  final String? category;
  final bool active;
  final String? source;
  final String? verificationStatus;

  AllergyIntolerance({
    required this.patient,
    required this.code,
    required this.display,
    required this.timestamp,
    this.criticality,
    this.severity,
    this.category,
    required this.active,
    this.source,
    this.verificationStatus,
  });

  factory AllergyIntolerance.fromJson(Map<String, dynamic> json) {
    return AllergyIntolerance(
      patient: json['patient'] as String,
      code: json['code'] as String,
      display: json['display'] as String,
      timestamp: json['timestamp'] as String,
      criticality: json['criticality'] as String?,
      severity: json['severity'] as String?,
      category: json['category'] as String?,
      active: json['active'] as bool,
      source: json['source'] as String?,
      verificationStatus: json['verificationStatus'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'patient': patient,
      'code': code,
      'display': display,
      'timestamp': timestamp,
      'criticality': criticality,
      'severity': severity,
      'category': category,
      'active': active,
      'source': source,
      'verificationStatus': verificationStatus,
    };
  }
}

File: server/Records/app/ax_types/allergyIntolerance.py
from enum import Enum
from typing import Optional
import strawberry
from typing import Optional,List

from app.ax_types.util import Json_type


@strawberry.enum
class Criticality(Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"


@strawberry.enum
class Severity(Enum):
    MILD = "mild"
    MODERATE = "moderate"
    SEVERE = "severe"



@strawberry.enum
class Source(Enum):
    PATIENTREPORTED = "patient-reported"
    DOCTORRECORDED = "physician-recorded"


@strawberry.enum
class Status(Enum):
    CONFIRMED = "confirmed"
    SUSPECTED = "suspected"



@strawberry.type
class AllergyIntolerance:
    patientID:Optional[str] = None
    code:Optional[str] = None
    display:Optional[str] = None
    timestamp:Optional[str] = None
    criticality:Optional[Criticality] = None
    severity:Optional[Severity] = None
    category:Optional[str]=None
    active:Optional[bool] = None
    source:Optional[Source] = None
    verificationStatus:Optional[Status] = None
    meta:Optional[Json_type]= None


@strawberry.type
class AllergyIntoleranceStack:
    allergyIntolerances :List[AllergyIntolerance] # type: ignore